src/clj/rems/application/commands.clj

```diff
@@ -566,6 +569,24 @@
   (ok {:event/type :application.event/external-id-assigned
        :application/external-id (:external-id cmd)}))
 
+(defn- forbidden-error [application cmd]
+  (let [permissions (if application
+                      (permissions/user-permissions application (:actor cmd))
+                      #{:application.command/create})]
+    (when-not (contains? permissions (:type cmd))
+      {:errors [{:type :forbidden}]})))
+
+(defmethod command-handler :application.command/delete
+  [cmd application injections]
+  ;; Just like in :application.command/create, we need to avoid
+  ;; accidental side-effects here. handle-command runs the
+  ;; command-handler even if the command is forbidden, so we duplicate
+  ;; the access control here.
```

Comment:
Uuh this is slightly smelly. I wonder if it would work to create a deleted event and then handle the effect in a process managed? I'm not saying this has to be changed.

Reply:
Yeah I was a bit hesitant about this too. I'll try the process manager thing out!

Reply:
It turned out rather nicely!